src/Service/ZGWToXxllncService.php

```diff
@@ -570,6 +661,70 @@ private function syncZaakToXxllnc(): array
     }//end syncZaakToXxllnc()
 
 
+    /**
+     * Maps the Besluit to a xxllnc case, post it, and creates a realtion to the actual case.
+     * We create a sub case for the Besluit because the xxllnc api does not have a Besluit variant.
+     */
+    private function syncBesluitToXxllnc()
+    {
+        if (isset($this->data['zaak']) === false) {
+            var_dump('syncBesluitToXxllnc returned, no zaak set');
+
+            return [];
+        }
+
+        if (isset($this->data['besluit']['besluittype']) === false) {
+            var_dump('syncBesluitToXxllnc returned, no zaak set');
+
+            return [];
+        }
+
+        $zaakObject      = $this->entityManager->find('App:ObjectEntity', $this->data['zaak']['_self']['id']);
+        $zaakArrayObject = $zaakObject->toArray();
+
+        $besluitTypeObject = $this->entityManager->find('App:ObjectEntity', $this->data['besluit']['besluittype']['_self']['id']);
+        if ($besluitTypeObject === null) {
+            var_dump('syncBesluitToXxllnc returned, no besluitType set on besluit');
+
+            return [];
+        }
+
+        // If set we already synced the Zaak to xxllnc as a case.
+        if ($besluitTypeObject->getSynchronizations() && $besluitTypeObject->getSynchronizations()->first() && $besluitTypeObject->getSynchronizations()->first()->getSourceId()) {
+            $besluitCaseTypeId = $besluitTypeObject->getSynchronizations()->first()->getSourceId();
+        } else {
+            var_dump('syncBesluitToXxllnc returned, the current besluittype had no source id so it did not came from the xxllnc api.');
+
+            return [];
+        }
+
+        // If set we already synced the Zaak to xxllnc as a case.
+        if ($zaakObject->getSynchronizations() && $zaakObject->getSynchronizations()->first() && $zaakObject->getSynchronizations()->first()->getSourceId()) {
+            $caseSourceId = $zaakObject->getSynchronizations()->first()->getSourceId();
+        }
+
+        // If the Zaak hasn't been send to xxllnc yet, do it now.
+        if (isset($caseSourceId) === false) {
+            // @TODO Lets expect the case has been synced already for now...
+            // @TODO Make it possible to sync zaak from here
+            // $this->syncZaakToXxllnc();
+        }
+
+        $this->mapBesluitToXxllnc($besluitCaseTypeId, $this->data['besluit'], $besluitCaseTypeId);
+
+        // Link normal case and besluit case at xxllnc api.
+        $this->sendCaseRelationForBesluit($caseSourceId, $besluitCaseSourceId);
+        var_dump('test syncBesluitToXxllnc');
+        die;
```

Comment:
Is this meant to die?

Reply:
These changes were not for this pr, so I reverted the file